src/nyx/main_nyxus.cpp

```diff
@@ -9,26 +9,49 @@
 	bool gpu_initialize(int dev_id); 
 #endif
 
+using namespace Nyxus;
+
+namespace Nyxus {
+	int processDataset(
+		const std::vector<std::string>& intensFiles, 
+		const std::vector<std::string>& labelFiles, 
+		int numFastloaderThreads, 
+		int numSensemakerThreads, 
+		int numReduceThreads, 
+		int min_online_roi_size, 
+		const SaveOption saveOption, 
+		const std::string& outputPath);
+
+	int processDataset_3D(
+		const std::vector <Imgfile3D_layoutA>& intensFiles,
+		const std::vector <Imgfile3D_layoutA>& labelFiles,
+		int numFastloaderThreads,
+		int numSensemakerThreads,
+		int numReduceThreads,
+		int min_online_roi_size,
+		const SaveOption saveOption,
+		const std::string& outputPath);
+};
+
```

Comment:
Again, those functions are called only in this file and their arguments were and may be in the future intensely changed (I may need a different type instead of Imgfile3D_layoutA), so why have the 3rd place (file globals.h) to edit the function signature and a full rebuild?